demisto_sdk/commands/common/update_id_set.py

```diff
@@ -1076,10 +1103,124 @@ def get_general_paths(path, pack_to_create):
     files = list()
     for path in path_list:
         files.extend(glob.glob(os.path.join(*path)))
+    print_color(f"files: {files}", LOG_COLORS.WHITE)
+
+    return files
+
+
+def get_object_entities_paths(path, pack_to_create):
+    """
+    get paths of objectTypes, objectFields
+
+    """
+    if pack_to_create:
+        path_list = [
+            [pack_to_create, path, '*', '*.json']
+        ]
+
+    else:
+        path_list = [
+            [path, '*'],
+            ['Packs', '*', path, '*', '*.json']
+        ]
+
+    files = list()
+    for path in path_list:
+        files.extend(glob.glob(os.path.join(*path)))
+    print_color(f"files: {files}", LOG_COLORS.WHITE)
 
     return files
 
 
+def get_object_type_data(path):
+    json_data = get_json(path)
+
+    id_ = json_data.get('id')
+    name = json_data.get('name', '')
+    fromversion = json_data.get('fromVersion')
+    toversion = json_data.get('toVersion')
+    playbook_id = json_data.get('playbookId')
+    pack = get_pack_name(path)
+    definitionId = json_data.get('definitionId')
+    layout = json_data.get('layout')
+
+    data = create_common_entity_data(path=path, name=name, to_version=toversion, from_version=fromversion, pack=pack)
+    if playbook_id and playbook_id != '':
+        data['playbooks'] = playbook_id
+
+    if definitionId:
+        data['definitionId'] = definitionId
+    if layout:
+        data['layout'] = layout
+    return {id_: data}
+
+
+def get_object_field_data(path, objects_types_list):
+    json_data = get_json(path)
+
+    id_ = json_data.get('id')
+    name = json_data.get('name', '')
+    fromversion = json_data.get('fromVersion')
+    toversion = json_data.get('toVersion')
+    pack = get_pack_name(path)
+    all_associated_types: set = set()
+    all_scripts = set()
+    definitionId = json_data.get('definitionId')
+
+    associated_types = json_data.get('associatedTypes')
+    if associated_types:
+        all_associated_types = set(associated_types)
+
+    system_associated_types = json_data.get('systemAssociatedTypes')
+    if system_associated_types:
+        all_associated_types = all_associated_types.union(set(system_associated_types))
+
+    if 'all' in all_associated_types:
+        all_associated_types = {list(object_type.keys())[0] for object_type in objects_types_list}
+
+    scripts = json_data.get('script')
+    if scripts:
+        all_scripts = {scripts}
+
+    field_calculations_scripts = json_data.get('fieldCalcScript')
+    if field_calculations_scripts:
+        all_scripts = all_scripts.union({field_calculations_scripts})
+
+    data = create_common_entity_data(path=path, name=name, to_version=toversion, from_version=fromversion, pack=pack)
+
+    if all_associated_types:
+        data['object_types'] = list(all_associated_types)
+    if all_scripts:
+        data['scripts'] = list(all_scripts)
```

Comment:
weird are we sure fields (incident fields included) can be related to a script? @Itay4 am I missing something?

Reply:
yea we have:
- on field change script
- field display script
- sla breach script